conda/cli/main.py

```diff
@@ -63,6 +63,34 @@ def init_loggers(context=None):
         set_verbosity(context.verbosity)
 
 
+def init_std_stream_encoding():
+    """
+    PY2 compat: Initialize encoding for standard streams.
+    Python 2 sets the encoding of stdout/stderr to None if not run in a
+    terminal context and thus falls back to ASCII.
+    """
+    for stream_name in ("stdout", "stderr"):
+        stream = getattr(sys, stream_name)
+        if not stream.encoding:
+            break
```

Comment:
This duplicate code is just to avoid the imports below if they are not neccessary.